src/rust/sysmon-subgraph-generator/src/main.rs

```diff
@@ -673,13 +703,16 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     };
 
     if grapl_config::is_local() {
-        let generator = SysmonSubgraphGenerator::new(NopCache {}, metrics);
```

Comment:
I could separate the `let generator = ...;` line out again if we want, but the main change is just calling `run_graph_generator` which should be best practice here. 

I really tried to find a way to do the following:


```
// ...

let cache = if grapl_config::is_local() {
    NopCache {}
} else {
    event_cache().await
};

run_graph_generator(SysmonSubgraphGenerator::new(cache, metrics), ZstdDecoder::default()).await;
```

But I was unable to figure out how to make Rust happy with that. If anyone knows of a way that doesn't look like complete gibberish or require some crazy type work, I'd love to see it. I hate this branching pattern.

My suspicion is we could use `Box<dyn Cache ...>` but I've not that much experience with `Box` and not sure if there is a way to pass the value of a `Box` into `SysmonSubgraphGenerator` without changing the type on the `:;new` fn.